File: apps/agent/src/paths.ts
import { homedir, hostname } from "node:os";
import { join } from "node:path";

export function getConfigDir(): string {
  return join(homedir(), ".bore");
}

export function getConfigPath(): string {
  return join(getConfigDir(), "config.json");
}

export function getRuntimePath(): string {
  return join(getConfigDir(), "runtime.json");
}

export function getDefaultDeviceName(): string {
  return hostname();
}

File: apps/agent/internal/agent/paths.go
package agent

import (
	"os"
	"path/filepath"
)

func getConfigDir() string {
	home, err := os.UserHomeDir()
	if err != nil {
		return ".bore"
	}

	return filepath.Join(home, ".bore")
}

func getConfigPath() string {
	return filepath.Join(getConfigDir(), "config.json")
}

func getRuntimePath() string {
	return filepath.Join(getConfigDir(), "runtime.json")
}
